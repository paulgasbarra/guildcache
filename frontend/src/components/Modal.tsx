import React from "react";
import { X } from "react-feather";

interface ModalProps {
  children: React.ReactElement | string;
  onClose: () => void;
  open: boolean;
}

const Modal: React.FC<ModalProps> = ({ children, onClose, open }) => {
  return (
    <div
      onClick={onClose}
      className={`fixed inset-0 flex justify-center items-center ${
        open ? "visible bg-black/20" : "invisible"
      }`}
    >
      <div
        onClick={(e) => e.stopPropagation()}
        className={`bg-white rounded-xl shadow p-10 transition-all ${
          open ? "scale-100 opacity-100" : "scale-125 opacity-0"
        }`}
      >
        <button
          id="close-modal"
          onClick={onClose}
          className="absolute top-2 right-2 p-1 rounded-lg text-gray-400 bg-white hover:bg-gray-50 hover:text-gray-600"
        >
          <X />
        </button>
        {children}
      </div>
    </div>
  );
};

export default Modal;
