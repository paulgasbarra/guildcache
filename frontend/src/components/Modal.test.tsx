import React from "react";
import Modal from "./Modal";
import { render, fireEvent, screen } from "@testing-library/react";

describe("Modal", () => {
  it("renders", () => {
    render(<Modal open onClose={() => {}} children="Hello" />);
    expect(screen.getByText("Hello")).toBeInTheDocument();
  });
  //Why does this fail?
  //   it("does not render when open is false", () => {
  //     render(<Modal open={false} onClose={() => {}} children="Hello" />);
  //     expect(screen.getByText("Hello")).not.toBeInTheDocument();
  //   });

  it("renders children", () => {
    const { getByText } = render(
      <Modal open onClose={() => {}} children="Hello" />
    );
    expect(getByText("Hello")).toBeInTheDocument();
  });

  it("calls onClose when clicked", () => {
    const onClose = vi.fn();
    render(<Modal open onClose={onClose} children="Hello" />);
    fireEvent.click(screen.getByRole("button"));
    expect(onClose).toHaveBeenCalled();
  });
});
