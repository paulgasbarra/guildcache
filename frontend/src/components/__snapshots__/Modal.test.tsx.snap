// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > does not render when open is false 1`] = `
<div>
  <div
    class="fixed inset-0 flex justify-center items-center invisible"
  >
    <div
      class="bg-white rounded-xl shadow p-10 transition-all scale-125 opacity-0"
    >
      <button
        class="absolute top-2 right-2 p-1 rounded-lg text-gray-400 bg-white hover:bg-gray-50 hover:text-gray-600"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
      Hello
    </div>
  </div>
</div>
`;

exports[`Modal > renders 1`] = `
<div>
  <div
    class="fixed inset-0 flex justify-center items-center visible bg-black/20"
  >
    <div
      class="bg-white rounded-xl shadow p-10 transition-all scale-100 opacity-100"
    >
      <button
        class="absolute top-2 right-2 p-1 rounded-lg text-gray-400 bg-white hover:bg-gray-50 hover:text-gray-600"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
      Hello
    </div>
  </div>
</div>
`;
